### PR TITLE
Dynamically fetch the Heron version for website creation

### PR DESCRIPTION
**Pull request by lucperkins**

### CODE REVIEW

**Comment by lucperkins** [comment]
@kramasamy Actually, hold off on shipping this. I'm realizing now that the script currently alphabetizes all of the `config.yaml` params, which we really don't want. I'll see if I can undo that behavior.

**Comment by objmagic** [comment]
This PR looks very promising. It solves [the problem](https://github.com/twitter/heron/pull/815/files/f9ba523479345d807c5a02af84c19bee6743f748#r64969974) by @kramasamy. 
Thanks!

**Comment by objmagic** [comment]
if I understand this PR correctly, one can put arbitrary version number in ``config.yaml`` and CI will update it, but CI is not able to commit the corrected ``config.yaml``. This means version number field is necessary and could be omitted on ``upstream``, but we cannot omit it locally since we need to build and view the website. This situation is awkward to me.

File: website/scripts/heron-version.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python
+
```
Comment (billonahill) [inline website/scripts/heron-version.sh]: Please provide some human-speak comment that gives an overview of what this script does.

Comment (lucperkins) [inline website/scripts/heron-version.sh]: @billonahill Done

**Comment by billonahill** [comment]
Thanks for clarifying. Really we want to have the release in the docs align with the git tag. So if you checkout a git tag for a release, the docs are properly labeled. With this approach they won't be. Could we instead update the release process to include changing the version in `config.yaml`? Either automatically via a release script (ideal) or manually via documented process.

**Comment by kramasamy** [comment]
@lucperkins  - if you can resolve the conflicts and we are ready to merge.

**Comment by billonahill** [comment]
@kramasamy I don't think we should merge this as is. See my comments above.

**Comment by lucperkins** [comment]
@billonahill How is this approach different from what you're proposing? This approach determines the current Heron version via Git tags and then updates the version value in the website config.

Whether this script is called during the broader Heron release process is up to others, but as for the `make site` process this seems to do precisely what you're calling for.

**Comment by billonahill** [comment]
The release tag in git should already include an updated config.yaml file to have a matching version.

**Comment by lucperkins** [comment]
@billonahill But the issue with that approach is that there's nothing to ensure that that happens. @kramasamy explicitly asked me to create a script that automates keeping the two in sync. To be honest, I don't really mind either way; I'm just taking cues from the project lead.

**Comment by billonahill** [comment]
Confirmed with @kramasamy that we will update the site version as part of the release process before we create the release tag. To be documented as part of #363. Closing this out.

**Pull request closed without merging**